src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java

```diff
@@ -179,4 +163,73 @@ public void updateLobby(ILobby lobby, String token, String lobbyName, Visibility
             lobby.setGameType(gameType);
         }
     }
+
+    public byte[] getQRCodeFromLobby(String token, Long lobbyId){
+        checkIfAuthenticationWasProvided(token);
+
+        ILobby lobby = getLobbyByIdElseThrowNotFound(lobbyId);
+
+        if(!checkUserIsInLobby(lobby, token)){
+            // If no user was found matching the token throw an error
+            String errorMessage = "The provided authentication was incorrect.";
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, errorMessage);
+        }
+
+        byte[] qrCode = lobby.getQrCode();
+
+        // If the QRCode is not already set generate it
+        if(qrCode == null){
+            try{
+                lobby.generateQrCode();
+            }catch (RestClientException e){
+                String errorMessage = "The server received an invalid response from the upstream server.";
+                throw new ResponseStatusException(HttpStatus.BAD_GATEWAY, errorMessage, e);
+            }
+            qrCode = lobby.getQrCode();
+        }
+
+        return qrCode;
+    }
+
+    private <T extends Enum> void checkEnumConfigNull(T config, String configName, String errorMessageEnding) {
+        if (config == null) {
+            String errorMessage = "The " + configName + " provided is empty. Therefore, the lobby could not be " + errorMessageEnding + "!";
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, errorMessage);
+        }
+    }
+
+    private void checkStringConfigNullOrEmpty(String s, String fieldName, String errorMessageEnding) {
+        if (s == null || s.trim().isEmpty()) {
+            String errorMessage = "The lobby " + fieldName + " provided is empty. Therefore, the lobby could not be " + errorMessageEnding + "!";
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, errorMessage);
+        }
+    }
+
+    private void checkIfAuthenticationWasProvided(String token) {
```

Comment:
I'm not such a big fan of the separation see my comment on:
https://github.com/sopra-fs22-group-16/sopra-fs22-group-16-server/pull/224#pullrequestreview-937120003

Reply:
Yes, I agree with most of your points but the method then handles two things. Also you have to declare everywhere that what you want to test is not a token...
I would also have to use that method for the get QRCode to check if the token is set and the last sentence for token doesn't make sense. We could do it like this:

`private void checkStringConfigNullOrEmpty(String s, String fieldName, String errorMessageEnding) {
    if(fieldName.equals("token")){
        // Check if authentication was provided
        if (s == null || s.isEmpty()) {
            String errorMessage = "The user needs to provide authentication to retrieve lobby information." + errorMessageEnding + "!";
            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, errorMessage);
        }
    }else{
        if (s == null || s.trim().isEmpty()) {
            String errorMessage = "The lobby " + fieldName + " provided is empty. Therefore, the lobby could not be " + errorMessageEnding + "!";
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, errorMessage);
        }
    }
}`

so the method uses the token when the field is called token, and you can set the second part of the token method, so that it matches the use case

Reply:
![image](https://user-images.githubusercontent.com/23397674/162573886-7d66a712-81c8-486b-ad3c-688cf9227aa0.png)
idk, why github can't display code normally -.-

Reply:
I just looked at your version again, "access" does work. But i still hate the istoken boolean ^^

Reply:
![image](https://user-images.githubusercontent.com/23397674/162574221-d9a4be96-35fc-42f6-b95e-6ace7a1c0cbf.png)
What do you think?

Reply:
You're right, isToken boolean is no elegant solution.
Hardcoded string comparison is also not optimal because as soon as we decide to rename "token" to "lobbyToken" or whatever your proposed solution fieldName.equals("token") will not match anymore ;)
But let's keep the string comparison whenever we stumble across a better solution we can apply it also here. The method works in a reusable manner and the chances are low that we rename our "token" 👍🏻